state/snapshotsManager.go

```diff
@@ -0,0 +1,454 @@
+	ChannelsProvider         IteratorChannelsProvider
+}
+
+type snapshotsManager struct {
+	isSnapshotInProgress     atomic.Flag
+	lastSnapshot             *snapshotInfo
+	shouldSerializeSnapshots bool
+	processingMode           common.NodeProcessingMode
+
+	stateMetrics         StateMetrics
+	marshaller           marshal.Marshalizer
+	addressConverter     core.PubkeyConverter
+	trieSyncer           AccountsDBSyncer
+	processStatusHandler common.ProcessStatusHandler
+	channelsProvider     IteratorChannelsProvider
+	accountFactory       AccountFactory
+	mutex                sync.RWMutex
+}
+
+// 	NewSnapshotsManager creates a new snapshots manager
+func NewSnapshotsManager(args ArgsNewSnapshotsManager) (*snapshotsManager, error) {
+	if check.IfNil(args.Marshaller) {
+		return nil, ErrNilMarshalizer
+	}
+	if check.IfNil(args.ProcessStatusHandler) {
+		return nil, ErrNilProcessStatusHandler
+	}
+	if check.IfNil(args.AddressConverter) {
+		return nil, ErrNilAddressConverter
+	}
+	if check.IfNil(args.StateMetrics) {
+		return nil, ErrNilStateMetrics
+	}
+	if check.IfNil(args.ChannelsProvider) {
+		return nil, ErrNilChannelsProvider
+	}
+	if check.IfNil(args.AccountFactory) {
+		return nil, ErrNilAccountFactory
+	}
+
+	return &snapshotsManager{
+		isSnapshotInProgress:     atomic.Flag{},
+		lastSnapshot:             &snapshotInfo{},
+		shouldSerializeSnapshots: args.ShouldSerializeSnapshots,
+		processingMode:           args.ProcessingMode,
+		stateMetrics:             args.StateMetrics,
+		marshaller:               args.Marshaller,
+		addressConverter:         args.AddressConverter,
+		trieSyncer:               nil, //this will be set later
+		processStatusHandler:     args.ProcessStatusHandler,
+		channelsProvider:         args.ChannelsProvider,
+		mutex:                    sync.RWMutex{},
+		accountFactory:           args.AccountFactory,
+	}, nil
+}
+
+// SetSyncer sets the given syncer as the syncer for the underlying trie
+func (sm *snapshotsManager) SetSyncer(syncer AccountsDBSyncer) error {
+	if check.IfNil(syncer) {
+		return ErrNilTrieSyncer
+	}
+
+	sm.mutex.Lock()
+	defer sm.mutex.Unlock()
+
+	sm.trieSyncer = syncer
+	return nil
+}
+
+// IsSnapshotInProgress returns true if a snapshot is in progress
+func (sm *snapshotsManager) IsSnapshotInProgress() bool {
+	return sm.isSnapshotInProgress.IsSet()
+}
+
+func (sm *snapshotsManager) getTrieSyncer() AccountsDBSyncer {
+	sm.mutex.RLock()
+	defer sm.mutex.RUnlock()
+
+	return sm.trieSyncer
+}
+
+// StartSnapshotAfterRestartIfNeeded starts a snapshot after restart if needed
+func (sm *snapshotsManager) StartSnapshotAfterRestartIfNeeded(trieStorageManager common.StorageManager) error {
+	if check.IfNil(sm.getTrieSyncer()) {
+		return ErrNilTrieSyncer
+	}
+
+	if !trieStorageManager.ShouldTakeSnapshot() {
+		return nil
+	}
+
+	rootHash, epoch, err := sm.getSnapshotRootHashAndEpoch(trieStorageManager)
+	if err != nil {
+		log.Debug("could not retrieve snapshot info", "error", err)
+		return nil
```

Comment:
Yes, we do not start the snapshot because we do not know which rootHash to snapshot and for which epoch. When the node will reach the next epoch, it will start the snapshot for that epoch.
Regarding import-db, the process will be the same. If the import-db is  stoped when a snapshot is in progress, that snapshot will restart when the node restarts.

Reply:
Changed the log to WARN.